src/transformers/models/deta/modeling_deta.py

```diff
@@ -244,7 +244,7 @@ class DetaObjectDetectionOutput(ModelOutput):
 
 
 def _get_clones(module, N):
-    return nn.ModuleList([copy.deepcopy(module) for i in range(N)])
+    return nn.ModuleList([module for i in range(N)])
```

Comment:
This enables the dynamic loading to work properly.

If this is left as-is, what happens, is that during model init, all n modules are created pointing to the same parameter, but after `load_state_dict` only the first layer gets updated